ontologies/sh/src/shacl-af.ttl

```diff
@@ -0,0 +1,230 @@
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
```

Comment:
Hello @holgerknublauch. I have pieced together terms from the SHACL-AF draft so that we can present them on prefix.zazuko.com and integrate in out toolset where we generate code and provide code completion for vocabularies.

Does the below look correct? I have not seen `sh:NodeExpression` mentioned explicitly but I figure it makes sense at least as class. Could also add SHACL constraints to the classes according to the comments.

Reply:
There are no types such as sh:NodeExpression, and they would never be instantiated as all node expressions are entirely controlled and distinguished by their properties. Also, any RDF literal or URI node are also node expressions, so rdfs:Resource rdfs:subClassOf sh:NodeExpression.

So I don't see value in these classes, but the properties can be useful. Have you considered writing shapes that can point at a node expression and validate the syntax? That would be of more value, e.g. to syntax check valid node expressions.

Reply:
Ok, so I will remove `sh:NodeExpression`. I did think to make other expression types shapes with implicit target. So considered refactoring all of them like:

```turtle
sh:ExistsExpression
  a rdfs:Class, sh:NodeShape ;
  sh:property [
    sh:path sh:exists ;
    sh:minCount 1 ;
    sh:maxCount 1 ;
    sh:class sh:Shape ;
  ] ;
.
```

Coming back to `sh:NodeExpression`, would, if at all, would you remodel `rdfs:range sh:NodeExpression` of most properties?

Reply:
I think this would be a nice use case of pure sh:NodeShapes that point at each other using sh:node.